test/unit/StakingPool/depositTo.js

```diff
@@ -0,0 +1,774 @@
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.be.revertedWith('StakingPool: The pool is private');
+
+    const coverSigner = await ethers.getImpersonatedSigner(cover.address);
+
+    await expect(
+      stakingPool.connect(coverSigner).depositTo([
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.not.be.revertedWith('StakingPool: The pool is private');
+
+    await expect(
+      stakingPool.connect(manager).depositTo([
+        {
+          amount,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.not.be.revertedWith('StakingPool: The pool is private');
+  });
+
+  it('reverts if deposit amount is 0', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { trancheId, tokenId, destination } = depositToFixture;
+
+    await expect(
+      stakingPool.connect(user).depositTo([
+        {
+          amount: 0,
+          trancheId,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.be.revertedWith('StakingPool: Insufficient deposit amount');
+  });
+
+  it('reverts if tranche id is not active', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination } = depositToFixture;
+
+    const { maxTranche } = await getTranches();
+
+    await expect(
+      stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId: maxTranche + 1,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.be.revertedWith('StakingPool: Requested tranche is not yet active');
+  });
+
+  it('reverts if requested tranche expired', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await expect(
+      stakingPool.connect(user).depositTo([
+        {
+          amount,
+          trancheId: firstActiveTrancheId - 1,
+          tokenId,
+          destination,
+        },
+      ]),
+    ).to.be.revertedWith('StakingPool: Requested tranche has expired');
+  });
+
+  it('mints a new nft if token is 0', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination, depositNftId } = depositToFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    {
+      const totalSupply = await stakingPool.totalSupply();
+      expect(totalSupply).to.equal(1);
+    }
+
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
```

Comment:
let's zero out destination so it just mints to user.

we're testing the "no defined destination" situation here vs the next test.

Reply:
The `destination` variable is set to `AddressZero` in the setup. So all tests use the "no defined destination" except for the next test where we validate a specific destination. Do you think the default destination value and naming are confusing and should be improved?

Reply:
I see, i missed the fact that destination = AddressZero

should be fine.